typing/typecore.ml

```diff
@@ -4114,22 +4124,22 @@ and type_let ?(check = fun s -> Warnings.Unused_var s)
               generalize_structure ty'
             end;
             let exp =
-              Builtin_attributes.with_warning_attribute pvb_attributes
+              Builtin_attributes.warning_scope pvb_attributes
```

Comment:
I would argue that pattern typing should also be done in this warning scope for let binding, in order to be able to write for instance
```OCaml
let[@warning "-8"] [] = []
```
i.e. update
```let (pat_list, new_env, force, unpacks) =
    type_pattern_list env spatl scope nvs allow in
```
at line 4012 to take in account the warning attributes on the let binding.

Reply:
Well spotted!  However, note that warning 8 is not processed when type-checking the pattern, but a bit later.  I've thus added two warning scopes, and made sure that ppwarning attributes are not processed several times because of that.